packages/components/addon/components/hds/dropdown/list-item.hbs

```diff
@@ -0,0 +1,84 @@
+{{#if (eq this.item "title")}}
+  <li class={{this.classNames}}>
+    {{@text}}
+  </li>
+
+{{else if (eq this.item "generic")}}
+  <li class={{this.classNames}}>
+    {{yield}}
+  </li>
+
+{{else if (eq this.item "copy-item")}}
+  <li class={{this.classNames}}>
+    {{#if @copyItemTitle}}
+      <div class="hds-dropdown-list-item--copy-item-title">{{@copyItemTitle}}</div>
+    {{/if}}
+    <button
+      type="button"
+      class="{{if @state (concat 'is-' @state)}} foo {{if this.isSuccess 'is-success'}}"
+      {{on "click" this.copyCode}}
+    >
+      <div class="hds-dropdown-list-item__copy-item-text">
+        {{@text}}
+      </div>
+      <FlightIcon
+        @name="{{if this.isSuccess 'clipboard-checked' 'clipboard-copy'}}"
+        class="hds-dropdown-list-item__copy-item-icon"
+      />
+    </button>
+  </li>
+
+{{else if (eq this.item "description")}}
+  <li class={{this.classNames}}>
+    {{@text}}
+  </li>
+
+{{else if (eq this.item "separator")}}
+  <li class={{this.classNames}} role="separator"></li>
+
+{{else if (eq this.item "interactive")}}
+  <li class={{this.classNames}}>
+    {{#if @route}}
+      <LinkTo
+        class="{{if @state (concat 'is-' @state)}}"
+        @current-when={{@current-when}}
+        @models={{hds-link-to-models @model @models}}
+        @query={{hds-link-to-query @query}}
+        @replace={{@replace}}
+        @route={{@route}}
+        ...attributes
+      >
+        {{#if @icon}}
+          <div class="hds-dropdown-list-item__interactive-icon">
+            <FlightIcon @name={{@icon}} @isInlineBlock={{false}} />
+          </div>
+        {{/if}}
+        <div class="hds-dropdown-list-item__interactive-text">
+          {{@text}}
+        </div>
+      </LinkTo>
+    {{else if @href}}
+      <a target="_blank" rel="noopener noreferrer" href={{@href}} class="{{if @state (concat 'is-' @state)}}">
+        {{#if @icon}}
+          <div class="hds-dropdown-list-item__interactive-icon">
+            <FlightIcon @name={{@icon}} @isInlineBlock={{false}} />
+          </div>
+        {{/if}}
+        <div class="hds-dropdown-list-item__interactive-text">
+          {{@text}}
+        </div>
+      </a>
+    {{else}}
+      <button class="{{if @state (concat 'is-' @state)}}" type="button" ...attributes>
```

Comment:
we have to pass ...attributes so they can add the {{on "click" this.something}} but we don't need it in the others. WDYT?

Reply:
I think that we have to document this (for the users), because is not obvious at all and they don't have a way to know how this works unless they read the documentation or look at the source code.